### ISSUE TITLE
[PROPOSAL] Flink integration should not fail when extra/custom flink confs are provided

### ISSUE THREAD

**Issue by athityakumar** [OpenLineage/OpenLineage#2281]
**Purpose:**

Currently, the `public static OpenLineageYaml loadOpenLineageYaml(InputStream inputStream)` method is used to parse the openlineage-related flink confs. However, unlike the openlineage-spark integration wherein any custom confs/keys don't raise an exception, openlineage-flink raises an exception for any custom keys as the YML parsing schema is strict & bound by `OpenLineageYaml` POJO.

**Proposed implementation**

Ideally, the behavior should be same as openlineage-spark integration. Providing any extra/custom flink confs should just be ignored if they're not supported by openlineage, and shouldn't raise an exception that fails the Flink job.

This could be achieved by adding the `@JsonIgnoreProperties(ignoreUnknown = true)` annotation to the `OpenLineageYaml` POJO class.

**Comment by pawel-big-lebowski** [comment]
What's the practical scenario behind that? 
I mean, if I put wrong config entries, I would love my to fail instead of silently proceeding with incorrect configuration.

**Comment by mobuchowski** [comment]
I agree with @athityakumar, it would be the desired behavior for me.

> I mean, if I put wrong config entries, I would love my app to fail immediately instead of silently proceeding with incorrect configuration.

Those don't have to be "wrong" or "incorrect", the fact that integration does not recognize some keys does not mean they are wrong.

**Comment by athityakumar** [comment]
@pawel-big-lebowski - Same point as @mobuchowski. And also, this is to have consistency / feature parity between the spark & flink integrations. We don't throw an error if we have unsupported extra openlineage spark-confs (like spark.openlineage.foo=bar), so I was thinking it'd be better to have the same experience in the flink integration as well

Certainly helps us with expanding support for custom confs too (related to https://github.com/OpenLineage/OpenLineage/issues/2273)
